kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KafkaRequestHandler.java

```diff
@@ -1213,25 +1253,58 @@ private void handleListOffsetRequestV1AndAbove(KafkaHeaderAndRequest listOffset,
                                                    CompletableFuture<AbstractResponse> resultFuture) {
         ListOffsetRequest request = (ListOffsetRequest) listOffset.getRequest();
 
-        Map<TopicPartition, CompletableFuture<ListOffsetResponse.PartitionData>> responseData = Maps.newHashMap();
-
-        request.partitionTimestamps().entrySet().stream().forEach(tms -> {
-            TopicPartition topic = tms.getKey();
-            Long times = tms.getValue();
-            CompletableFuture<ListOffsetResponse.PartitionData> partitionData;
-
-            partitionData = fetchOffsetForTimestamp(KopTopic.toString(topic), times, false);
+        List<CompletableFuture<Pair<TopicPartition, ListOffsetResponse.PartitionData>>>
+                allFutures = new ArrayList<>(request.partitionTimestamps().size());
+        Map<TopicPartition, ListOffsetResponse.PartitionData> responseMap = Maps.newConcurrentMap();
 
-            responseData.put(topic, partitionData);
+        request.partitionTimestamps().forEach((topic, times) -> {
+            CompletableFuture<Pair<TopicPartition, ListOffsetResponse.PartitionData>> future =
+                    new CompletableFuture<>();
+            String fullPartitionName = KopTopic.toString(topic);
+            authorize(AclOperation.DESCRIBE, Resource.of(ResourceType.TOPIC, fullPartitionName))
```

Comment:
`LIST_OFFSETS` request can be sent by consumer or admin. I found the implementation of `SimpleAclAuthorizer#canLookupAsync` authorizes `AuthAction.produce` first. If a consumer with only consume permission, the first check for produce permission will be redundant.

My suggested change is to check `AuthAction.consume` first in `SimpleAclAuthorizer#canLookupAsync`. If you have better idea, you can also apply it.